pkg/strconv/strconv.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2019 Tom Artale. All rights reserved.
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file.
+//
+// Package strconv implements conversions to and from string representations
+// of basic data types.
+package strconv
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// ParsePrimitive interprets a string s and saves the result
+// in the value pointed to by primitivePtr.
+//
+// This is a wrapper for all the Parse* functions provided
+// in strconv, but is useful when you're working with an
+// interface{} type and don't want to do a giant switch
+// statement. You trade-off the type-safety provided by
+// the existing functions.
+//
+// The base can be provided for the integer-like Parse functions,
+// if it is required. If the type of primitivePtr is not an
+// integer, the base is ignored.  If the base is not provided,
+// it defaults to 0.
+//
+// The bitSize argument required by some of the Parse functions
+// is provided automatically, depending on the type.
+//
+// Any errors from the underlying Parse functions are reflected
+// back to the caller, and the variable is left unchanged.
+func ParsePrimitive(s string, primitivePtr interface{}, optionalBase ...int) error {
+	base := 0
+	if len(optionalBase) == 1 {
+		base = optionalBase[0]
+	} else if len(optionalBase) > 1 {
+		return fmt.Errorf("Unexpected optional parameters: %v", optionalBase)
```

Comment:
Would you want to consider just telling the user that only one base can be provided here? The current error message seems a bit vague.